static/src/javascripts/projects/common/modules/discussion/comments.js

```diff
@@ -0,0 +1,588 @@
+// @flow
+
+import $ from 'lib/$';
+import bean from 'bean';
+import bonzo from 'bonzo';
+import qwery from 'qwery';
+import config from 'lib/config';
+import fetchJson from 'lib/fetch-json';
+import mediator from 'lib/mediator';
+import reportError from 'lib/report-error';
+import { constructQuery } from 'lib/url';
+import Component from 'common/modules/component';
+import DiscussionApi from 'common/modules/discussion/api';
+import { CommentBox } from 'common/modules/discussion/comment-box';
+import WholeDiscussion from 'common/modules/discussion/whole-discussion';
+import relativedates from 'common/modules/ui/relativedates';
+import userPrefs from 'common/modules/user-prefs';
+import { inlineSvg } from 'common/views/svgs';
+
+type userType = {
+    badge: Array<{
+        name: string,
+    }>,
+    displayName: string,
+    isStaff?: boolean,
+    privateFields: {
+        canPostComment: boolean,
+    },
+};
+
+type commentType = {
+    id: string,
+    body: string,
+};
+
+const PREF_RELATIVE_TIMESTAMPS: string = 'discussion.enableRelativeTimestamps';
+
+const shouldMakeTimestampsRelative = (): boolean =>
+    userPrefs.get(PREF_RELATIVE_TIMESTAMPS) !== null
+        ? userPrefs.get(PREF_RELATIVE_TIMESTAMPS)
+        : true;
+
+class Comments extends Component {
+    constructor(options: Object): void {
+        super();
+
+        this.componentClass = 'd-comments';
+        this.comments = null;
+        this.topLevelComments = null;
+        this.user = null;
+        this.classes = {
+            comments: 'd-thread--top-level',
+            topLevelComment: 'd-comment--top-level',
+            reply: 'd-comment--response',
+            showReplies: 'd-show-more-replies',
+            showRepliesButton: 'd-show-more-replies__button',
+            newComments: 'js-new-comments',
+            comment: 'd-comment',
+            commentReply: 'd-comment__action--reply',
+            commentPick: 'd-comment__action--pick',
+            commentStaff: 'd-comment--staff',
+            commentBody: 'd-comment__body',
+            commentTimestampJs: 'js-timestamp',
+            commentReport: 'js-report-comment',
+        };
+        this.defaultOptions = {
+            discussionId: null,
+            showRepliesCount: 3,
+            commentId: null,
+            order: 'newest',
+            threading: 'collapsed',
+        };
+
+        this.setOptions(options);
+    }
+
+    getMoreReplies(event: Event): void {
+        event.preventDefault();
+
+        const target: HTMLElement = (event.target: any);
+        const currentTarget: HTMLElement = (event.currentTarget: any);
+        const li = currentTarget.closest(this.getClass('showReplies'));
+
+        if (li) {
+            li.innerHTML = 'Loading…';
+        }
+
+        const source = target.getAttribute('data-source-comment');
+        const commentId = currentTarget.getAttribute('data-comment-id') || '';
```

Comment:
I see you're assigning `commentId` the value of an empty string here if the`getAttribute('data-comment-id') ` returns `null`. I know this is because flow would have been complaining that we were using `commentId` afterwards without handling the . _possible_ scenario it is `null`. 

Perhaps we should just wrap the code that uses `commentId` in an `if (commentId) { }` rather than assigning an empty string? This'll save us trying to fetchJson from the `endpoint`, which presumably returns nothing useful when `commentId` is an empty string?  It looks like you've done something similar on line 224.